caffe2/CMakeLists.txt

```diff
@@ -89,7 +89,7 @@ if (NOT INTERN_BUILD_MOBILE OR BUILD_CAFFE2_MOBILE)
       add_subdirectory(quantization)
       add_subdirectory(quantization/server)
     endif()
-    if (USE_QNNPACK)
+    if (USE_QNNPACK AND BUILD_CAFFE2_MOBILE)
```

Comment:
Important change, please double check.  Enable caffe2/quantized only when BUILD_CAFFE2_MOBILE is set.

Reply:
operators/quantized is only built for server or libcaffe2 mobile (line 67) - but for both cases BUILD_CAFFE2_MOBILE is true (right, it's default to true and we don't unset it for non-mobile build). The additional condition should be no-op - so what case are you trying to protect here?

In general, I saw you already added source code macro USE_PYTORCH_QNNPACK, maybe we can do the same thing in cmake, i.e. instead of changing cmake USE_QNNPACK's behavior with BUILD_CAFFE2_MOBILE macro, I'm thinking of a more decoupled setup:
- doesn't change cmake USE_QNNPACK logic;
- add USE_PYTORCH_QNNPACK cmake macro for cloned pytorch_qnnpack. For now we can simply copy USE_QNNPACK cmake logic - we should gradually refactor pytorch_qnnpack/CMakeList to make it more natively integrated with aten;
- for caffe2 mobile build, always set USE_PYTORCH_QNNPACK to false; USE_QNNPACK is still a user facing macro;
- for pytorch mobile build, always set USE_QNNPACK to false (as we don't build caffe2 ops); and always set USE_PYTORCH_QNNPACK to true - unless cmake/Dependencies.cmake logic determines it's not supported (check places where set USE_QNNPACK off currently).

Maybe we can call it INTERN_USE_PYTORCH_QNNPACK to keep it a private flag as eventually it will be part of aten natively. I think we can always build it unless the architecture is not supported by qnnpack implementation. Does it sound reasonable or not?